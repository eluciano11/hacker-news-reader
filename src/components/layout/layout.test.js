import React from "react";
import { render, cleanup } from "@testing-library/react";

import Layout from "./layout";

afterEach(cleanup);

describe("Layout Component Tests", () => {
  it("Should render layout with online mode", () => {
    const { asFragment } = render(<Layout isOnline={true}>Layout test</Layout>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should render layout with offline mode", () => {
    const { asFragment } = render(
      <Layout isOnline={false}>Layout test</Layout>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
