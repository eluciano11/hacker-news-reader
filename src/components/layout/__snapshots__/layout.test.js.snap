// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component Tests Should render layout with offline mode 1`] = `
<DocumentFragment>
  <main>
    <header
      class="header"
    >
      <div
        class="container"
      >
        <img
          alt="logo"
          class="logo"
          src="./yc.png"
        />
        <div
          class="title"
        >
          Hacker News Reader
        </div>
      </div>
      <div
        class="container"
      >
        <img
          alt="wifi-icon"
          class="networkStateIcon"
          src="./no-wifi.svg"
        />
        <p
          class="networkState"
        >
          Offline
        </p>
      </div>
    </header>
    Layout test
  </main>
</DocumentFragment>
`;

exports[`Layout Component Tests Should render layout with online mode 1`] = `
<DocumentFragment>
  <main>
    <header
      class="header"
    >
      <div
        class="container"
      >
        <img
          alt="logo"
          class="logo"
          src="./yc.png"
        />
        <div
          class="title"
        >
          Hacker News Reader
        </div>
      </div>
      <div
        class="container"
      >
        <img
          alt="wifi-icon"
          class="networkStateIcon"
          src="./wifi.svg"
        />
        <p
          class="networkState"
        >
          Online
        </p>
      </div>
    </header>
    Layout test
  </main>
</DocumentFragment>
`;
